glide get github.com/Azure/go-autorest/autorest fails

Steps:
* Removed `glide.lock`.
* Removed `vendor/`.
* Execute `glide get github.com/Azure/go-autorest/autorest`

Expected, it works.

Actual, it fails:

```
$ glide get github.com/Azure/go-autorest/autorest
[INFO] Preparing to install 1 package.
[INFO] Importing github.com/Azure/go-autorest
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/Azure/azure-sdk-for-go.
[INFO] Fetching updates for github.com/golang/glog.
[INFO] Fetching updates for github.com/spf13/cobra.
[INFO] Fetching updates for github.com/pborman/uuid.
[INFO] Fetching updates for github.com/Azure/go-autorest.
[INFO] Resolving imports
[INFO] Fetching updates for github.com/spf13/pflag.
[INFO] Fetching updates for github.com/Azure/go-autorest.
[INFO] Fetching updates for golang.org/x/crypto.
[INFO] Fetching updates for github.com/dgrijalva/jwt-go.
[ERROR] Error scanning github.com/Azure/go-autorest/azure: open /home/cole/code/azure/azkube_gopath/src/github.com/colemickens/azkube/vendor/github.com/Azure/go-autorest/azure: no such file or directory
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/Azure/azure-sdk-for-go.
[INFO] Fetching updates for github.com/pborman/uuid.
[INFO] Fetching updates for github.com/golang/glog.
[INFO] Fetching updates for github.com/spf13/cobra.
An Error has occured
```

I don't know where `glide` came up with this import path in the ERROR line. It's simply wrong: `github.com/Azure/go-autorest/azure`. (Note it's missing the `autorest` segment between `go-autorest` and `azure`.)

It seems to have gotten confused, I'm guessing because the "top package" is not at the top of the repo.

`glide` is from whatever is on master at this very instant (bff8b50d232e5005b452de18d54ca7439c596415).

Any advice for unblocking would be greatly appreciated.

--- a/dependency/resolver.go
+++ b/dependency/resolver.go
@@ -2,6 +2,7 @@ package dependency
 
 import (
 	"container/list"
+	"errors"
 	"runtime"
 	"sort"
 	//"go/build"
@@ -502,8 +503,9 @@ func (r *Resolver) resolveImports(queue *list.List, testDeps, addTest bool) ([]s
 				continue
 			}
 		} else if err != nil {
+			errStr := err.Error()
 			msg.Debug("ImportDir error on %s: %s", r.Handler.PkgPath(dep), err)
-			if strings.HasPrefix(err.Error(), "no buildable Go source") {
+			if strings.HasPrefix(errStr, "no buildable Go source") {
 				msg.Debug("No subpackages declared. Skipping %s.", dep)
 				continue
 			} else if os.IsNotExist(err) && !foundErr && !foundQ {
@@ -530,6 +532,16 @@ func (r *Resolver) resolveImports(queue *list.List, testDeps, addTest bool) ([]s
 					// see if this is on GOPATH and copy it?
 					msg.Info("Not found in vendor/: %s (1)", dep)
 				}
+			} else if strings.Contains(errStr, "no such file or directory") {
+				r.hadError[dep] = true
+				msg.Err("Error scanning %s: %s", dep, err)
+				msg.Err("This error means the referenced package was not found.")
+				msg.Err("Missing file or directory errors usually occur when multiple packages")
+				msg.Err("share a common dependency and the first reference encountered by the scanner")
+				msg.Err("sets the version to one that does not contain a subpackage needed required")
+				msg.Err("by another package that uses the shared dependency. Try setting a")
+				msg.Err("version in your glide.yaml that works for all packages that share this")
+				msg.Err("dependency.")
 			} else {
 				r.hadError[dep] = true
 				msg.Err("Error scanning %s: %s", dep, err)
@@ -576,10 +588,10 @@ func (r *Resolver) resolveImports(queue *list.List, testDeps, addTest bool) ([]s
 					r.VersionHandler.SetVersion(imp, addTest)
 				} else if err != nil {
 					r.hadError[dep] = true
-					msg.Warn("Error looking for %s: %s", imp, err)
+					msg.Err("Error looking for %s: %s", imp, err)
 				} else {
 					r.hadError[dep] = true
-					msg.Info("Not found: %s (2)", imp)
+					msg.Err("Not found: %s (2)", imp)
 				}
 			case LocGopath:
 				msg.Debug("Found on GOPATH, not vendor: %s", imp)
@@ -594,6 +606,11 @@ func (r *Resolver) resolveImports(queue *list.List, testDeps, addTest bool) ([]s
 			}
 		}
 
+	}
+
+	if len(r.hadError) > 0 {
+		// Errors occured so we return.
+		return []string{}, errors.New("Error resolving imports")
 	}
 
 	// FIXME: From here to the end is a straight copy of the resolveList() func.

--- a/dependency/resolver_test.go
+++ b/dependency/resolver_test.go
@@ -46,6 +46,8 @@ func TestResolveLocalDeep(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	h := &DefaultMissingPackageHandler{Missing: []string{}, Gopath: []string{}, Prefix: "../vendor"}
+	r.Handler = h
 
 	l, _, err := r.ResolveLocal(true)
 	if err != nil {
@@ -62,6 +64,8 @@ func TestResolve(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	h := &DefaultMissingPackageHandler{Missing: []string{}, Gopath: []string{}, Prefix: "../vendor"}
+	r.Handler = h
 
 	base := filepath.Join(os.Getenv("GOPATH"), "src/github.com/Masterminds/glide/vendor")
 	l, err := r.Resolve("github.com/codegangsta/cli", base)
@@ -91,6 +95,8 @@ func TestResolveAll(t *testing.T) {
 	if err != nil {
 		t.Fatalf("No new resolver: %s", err)
 	}
+	h := &DefaultMissingPackageHandler{Missing: []string{}, Gopath: []string{}, Prefix: "../vendor"}
+	r.Handler = h
 	l, err := r.ResolveAll(deps, false)
 	if err != nil {
 		t.Fatalf("Failed to resolve: %s", err)

